### ISSUE TITLE
@note with @parblock within a table data cell breaks the HTML output

### ISSUE THREAD

**Issue by eepp** [doxygen/doxygen#8734]
**Describe the bug**

When you put a `@note` command with multiple paragraphs between `@parblock` and `@endparblock`, and when such `@note` command is within a table data cell (`<td>`), what follows the note is outside the table.

**Expected behavior**

What follows the note remains within the cell.

**Screenshots**

This is a screenshot of the HTML rendering of the attached reproducer:

![image](https://user-images.githubusercontent.com/1816705/130802861-e7d65655-a704-4ed7-8862-2d576b115483.png)

As you can see, the last paragraph, which is after the note, is outside the table.

**To Reproduce**

Unzip the attachment ([doxygen-bug-note-in-td.zip](https://github.com/doxygen/doxygen/files/7047446/doxygen-bug-note-in-td.zip)) and run:

```
$ doxygen Doxyfile
```

Open `output/html/index.html`.

**Version**

Doxygen 1.9.2

```
LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
```

**Comment by albert-github** [comment]
The `@note` nicely shows the problem but is not the cause of the problem.
It looks like a `@endparblock` terminates the table which is shown in the test1.md output of the attached example and it is even more obvious from the output of the test2.md.
When **test2.md** is:
```
<table>
  <tr>
    <td>
      This1

      @parblock
      This2
      @endparblock

      This3
    </td>
    <td>
      This1

      @parblock
      This2
      @endparblock

      This3
    </td>
  </tr>
</table>
```
the corresponding output is:

![image](https://user-images.githubusercontent.com/5223533/130823835-73ed3a4c-f256-4afd-b0f4-9f084d2c8b41.png)


Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/7048308/example.tar.gz)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
